lib/api/core/entrypoints/embedded/index.js

```diff
@@ -0,0 +1,464 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  Bluebird = require('bluebird'),
+  Context = require('./context'),
+  debug = require('../../../../kuzzleDebug')('kuzzle:entry-point:embedded'),
+  EntryPoint = require('../entrypoint'),
+  fs = require('fs'),
+  http = require('http'),
+  HttpProtocol = require('./protocols/http'),
+  {
+    InternalError: KuzzleInternalError
+  } = require('kuzzle-common-objects').errors,
+  WebSocketProtocol = require('./protocols/websocket'),
+  SocketIoProtocol = require('./protocols/socketio'),
+  moment = require('moment'),
+  path = require('path'),
+  RequestContext = require('kuzzle-common-objects').models.RequestContext,
+  winston = require('winston'),
+  WinstonElasticsearch = require('winston-elasticsearch'),
+  WinstonSyslog = require('winston-syslog');
+
+class EmbeddedEntryPoint extends EntryPoint {
+  constructor (kuzzle) {
+    super(kuzzle);
+
+    this.kuzzle = kuzzle;
```

Comment:
Nitpicking: `this.kuzzle` is already set in superclass constructor, no need to set here again.